src/shared/cpu-set-util.c

```diff
@@ -82,6 +82,23 @@ char *cpu_set_to_range_string(const CPUSet *set) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char* cpu_set_to_mask_string(const CPUSet *a) {
+        _cleanup_free_ char *str = NULL;
+        unsigned mask = 0;
```

Comment:
so what does the kernel actually expect as a type for this? i mean, `unsigned` is on all relevant archs 32bit. 32 cpus don't seem so much. what does the kernel internally use? isn't it using at least a u64? or maybe even arbitrary size bitmap?

Reply:
Kernel expect hexadecimal bitmap mask. I've updated this, borrowing from util-linux cpuset.c.